examples/david_map_reduce_weighted.py

```diff
@@ -81,6 +81,8 @@ def downsample_opencv(image_path: str, downsampling_factor: int) -> np.ndarray:
 
     # Calculate new dimensions based on downsampling factor
     scale_factor = 1 / downsampling_factor
+    if img is None:
+        raise ValueError("'img' cannot be 'None'.")
```

Comment:
I think this should be a `RuntimeError` instead:

```suggestion
        raise RuntimeError("'img' is unexpectedly 'None'.")
```

Alternatively we could look to figure out the issue? From a quick scan online it looks like `cv2.imread` returns `None` in cases where the image couldn't be read. Can we parse a warning or a log somewhere to give the user better feedback?

Also, this looks like unexpected and rare behaviour. Perhaps we should be placing the code below in a `try/except` block and capturing the type error then? Unless this is more to appease typing?

Finally, is it worth adding some test cases to trigger this behaviour?, perhaps by passing corrupted or missing files?